tests/test.py

```diff
@@ -6173,6 +6165,41 @@ def testServerRead(self):
         super().testServerRead()
         self.assertIn("Finished processing client's connection", self.read_log(self.acra))
 
+class TestAcraIgnoresLegacyKeys(AcraCatchLogsMixin, BaseTestCase):
+    """
+    Ensure AcraServer won't exit with error when started with flags for key caching,
+    while keystore contains legacy keys (Connector<->Server, Connector<->Translator)
+    """
+
+    def checkSkip(self):
+        if KEYSTORE_VERSION != 'v1':
+            self.skipTest("test only for keystore v1")
+
+    def setUp(self):
+        super().setUp()
+
+        def create_key_file(name):
+            open(f"{KEYS_FOLDER.name}/{name}", "w").close()
+
+        try:
+            create_key_file('testclientid')
```

Comment:
with such test's template we test only `testclientid` key because of acra will fail on first error.
But we should be sure that we can work with each of these keys. So let's move all these logic from setUp to test function and iterate over all these types:
```
def setUp(self):
 # do nothing, we don't need acra process, engines and all other stuff
  pass

def testKeysCachedSuccessfully(self):
  with tempfile.TemporaryDirectory() as folder:
     for key in ('testclientid.pub', testclientid_server', ...):
        with self.assertRaises(Exception) as exc:
          create_key_file(key)
          log_file = tempfile.TemporaryFile()
          args = {'keystore_cache_size': 0, 'keystore_cache_on_start_enable': 'true', 'log_to_file': log_file.name, 'keys_dir': folder}
          self.fork_acra(**args)

      # use correct sub-field for msg and exception, it's only example
      self.assertEqual(exc.exception.msg, 'can't wait connection')
      with open(log_file.name, 'r') as f:
        logs = f.read()
      self.assertIn("Cached keystore on start successfully".lower(), logs)
      # remove invalid key to not raise same error at next iteration
      os.remove(key)
```

With such test's template we check all these key types and verify them separately.

Reply:
oops, my fault. misunderstood that we check correct startup with all of these keys... but my comment about opposite case) nevermind